On windows, construct environment separately when storing to registry

sbc100 opened the pull request:
We can't use the set of environment variable when setting local
valus and when storing to the registry.

This is because the new PATH value is constructed very differently
when updating the local shell compared to when updating the
registry.

Fixes: #634

aminya reviewed (changes requested):
See https://github.com/emscripten-core/emsdk/pull/642#discussion_r515405800. That PR considers all the cases (`--system`, `--permamnet`, `none of them`). It includes the tests for all of them.

aminya reviewed (changes requested):
I am interested to know if this fixes the Process PATH preservation.

You can add process test by pasting this code at [here](https://github.com/aminya/emsdk/blob/9b6acc9ee7cd917b196dd826c845010b34120fff/scripts/test_path_preservation.ps1#L81) and [here](https://github.com/aminya/emsdk/blob/9b6acc9ee7cd917b196dd826c845010b34120fff/scripts/test_path_preservation.ps1#L90)
```ps1
        if (Compare-Object -ReferenceObject $PATH_Process_BEFORE.Split(';') -DifferenceObject $PATH_Process.Split(';') ) {
            echo "Old process path is ...................."
            echo $PATH_Process_BEFORE
            echo "Current process path is ................"
            echo $PATH_Process
            throw "User PATH are changed while --system had been provided"
        }
```

sbc100 commented:
> I am interested to know if this fixes the Process PATH preservation.
> 
> You can add process test by pasting this code at [here](https://github.com/aminya/emsdk/blob/9b6acc9ee7cd917b196dd826c845010b34120fff/scripts/test_path_preservation.ps1#L81) and [here](https://github.com/aminya/emsdk/blob/9b6acc9ee7cd917b196dd826c845010b34120fff/scripts/test_path_preservation.ps1#L90)
> 
> ```powershell
>         if (Compare-Object -ReferenceObject $PATH_Process_BEFORE.Split(';') -DifferenceObject $PATH_Process.Split(';') ) {
>             echo "Old process path is ...................."
>             echo $PATH_Process_BEFORE
>             echo "Current process path is ................"
>             echo $PATH_Process
>             throw "User PATH are changed while --system had been provided"
>         }
> ```

I don't think that will work because activation *is* supposed to add to the process path so it will be different afterwards.  

The test that I've added added in this RR in `scripts/check_path.py` instread checked that all of the element of the original path still appear in the new path and that none were removed.  i.e. that the operation only added to the process path and did not remove or re-order the existing process path.
